nix-tools-release.nix

```diff
@@ -0,0 +1,69 @@
+commonLib: # the iohk-nix commonLib
+{ packages ? []
+, required-name ? "required"
+, required-targets ? (jobsets: [])
+, config ? {}
+, rev ? "abcdef"
+, package-set-path # usually import ./. {}
+}:
+{ system ? builtins.currentSystem
+, pkgs ? commonLib.getPkgs { inherit system config; }
+
+, scrubJobs ? true
+, supportedSystems ? [ "x86_64-linux" "x86_64-darwin" ]
+, nixpkgsArgs ? {
+    config = config // { allowUnfree = false; inHydra = true; };
+  }
+}:
+with (import (commonLib.nixpkgs + "/pkgs/top-level/release-lib.nix") {
+  inherit supportedSystems scrubJobs nixpkgsArgs;
+  packageSet = import package-set-path;
+});
+with pkgs.lib;
+let
+
+  traceId = x: builtins.trace (builtins.deepSeq x x) x;
+
+  packageSet = import package-set-path {};
+  nix-tools-pkgs = supportedSystems: {
+    nix-tools.libs =
+      mapAttrs (_: _: supportedSystems)
+        (filterAttrs (n: v: builtins.elem n packages && v != null) packageSet.nix-tools.libs);
+    # aggreated exes
+    nix-tools.exes =
+      mapAttrs (_: _: supportedSystems)
+        (filterAttrs (n: v: builtins.elem n packages && v != null) packageSet.nix-tools.exes);
+    # component exes exposed
+    nix-tools.cexes =
+      mapAttrs (_: mapAttrs (_: _: supportedSystems))
+        (filterAttrs (n: v: builtins.elem n packages && v != null) packageSet.nix-tools.cexes);
+    nix-tools.tests =
+      mapAttrs (_: mapAttrs (_: _: supportedSystems))
+        (filterAttrs (n: v: builtins.elem n packages && v != null) packageSet.nix-tools.tests);
+    nix-tools.benchmarks =
+      mapAttrs (_: mapAttrs (_: _: supportedSystems))
+        (filterAttrs (n: v: builtins.elem n packages && v != null) packageSet.nix-tools.benchmarks);
+  };
+
+  mapped-pkgs = mapTestOn (nix-tools-pkgs supportedSystems);
+  # we use builtins.currentSystem here as that will evaluate to whatever the evaluator runs on.
+  # thus someone on macOS will be able to build the .x86_64-darwin cross expressions, while
+  # someone on linux will be able to build the .x86_64-linux ones.  As hydra is running on
+  # linux, this should also only present CI with the .x86_64-linux targets.
+  mapped-pkgs-mingw32 = mapTestOnCross lib.systems.examples.mingwW64 (nix-tools-pkgs [ builtins.currentSystem ]);
+
+  mapped-pkgs-all
+    = lib.recursiveUpdate
+        (mapped-pkgs)
+        (lib.mapAttrs (_: (lib.mapAttrs (_: (lib.mapAttrs' (n: v: lib.nameValuePair (lib.systems.examples.mingwW64.config + "-" + n) v)))))
+          mapped-pkgs-mingw32);
+
+in fix (self: (builtins.removeAttrs packageSet ["nix-tools" "_lib"]) // mapped-pkgs-all
+// {
+  forceNewEval = pkgs.writeText "forceNewEval" rev;
```

Comment:
I guess this is so you can force re-evaluation by changing the rev?

Reply:
I believe that's the idea, this logic was in the release.nix of cardano-chain, and I think I've also seen it in cardano-sl; thus I've just tried to keep it they way it was.